foyer-memory/src/cache.rs

```diff
@@ -226,64 +220,177 @@ where
 {
-    fn from(value: LfuCacheConfig<K, V, L, S>) -> Self {
-        Self::Lfu(value)
+    pub fn new(capacity: usize) -> Self {
+        Self {
+            capacity,
+            shards: None,
+            eviction_config: None,
+            object_pool_capacity: None,
+            event_listener: DefaultCacheEventListener::default(),
+            hash_builder: RandomState::default(),
+            _marker: PhantomData,
+        }
     }
 }
 
-impl<K, V, L, S> From<S3FifoCacheConfig<K, V, L, S>> for CacheConfig<K, V, L, S>
+impl<K, V, L, S> CacheBuilder<K, V, L, S>
 where
     K: Key,
     V: Value,
     L: CacheEventListener<K, V>,
     S: BuildHasher + Send + Sync + 'static,
 {
-    fn from(value: S3FifoCacheConfig<K, V, L, S>) -> Self {
-        Self::S3Fifo(value)
+    const DEFAULT_SHARDS: usize = 1;
+    const DEFAULT_EVICTION_CONFIG: EvictionConfig = EvictionConfig::Lfu(LfuConfig {
+        window_capacity_ratio: 0.1,
+        protected_capacity_ratio: 0.8,
+        cmsketch_eps: 0.001,
+        cmsketch_confidence: 0.9,
+    });
+    const DEFAULT_OBJECT_POOL_CAPACITY_RATIO_RECIPROCAL: usize = 10;
+
+    pub fn with_shards(mut self, shards: usize) -> Self {
+        self.shards = Some(shards);
+        self
+    }
+
+    pub fn with_eviction_config(mut self, eviction_config: impl Into<EvictionConfig>) -> Self {
+        self.eviction_config = Some(eviction_config.into());
+        self
+    }
+
+    pub fn with_object_pool_capacity(mut self, object_pool_capacity: usize) -> Self {
+        self.object_pool_capacity = Some(object_pool_capacity);
+        self
+    }
+
+    pub fn with_event_listener<OL>(self, event_listener: OL) -> CacheBuilder<K, V, OL, S>
+    where
+        OL: CacheEventListener<K, V>,
+    {
+        CacheBuilder {
+            capacity: self.capacity,
+            shards: self.shards,
+            eviction_config: self.eviction_config,
+            object_pool_capacity: self.object_pool_capacity,
+            event_listener,
+            hash_builder: self.hash_builder,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn with_hash_builder<OS>(self, hash_builder: OS) -> CacheBuilder<K, V, L, OS>
+    where
+        OS: BuildHasher + Send + Sync + 'static,
+    {
+        CacheBuilder {
+            capacity: self.capacity,
+            shards: self.shards,
+            eviction_config: self.eviction_config,
+            object_pool_capacity: self.object_pool_capacity,
+            event_listener: self.event_listener,
+            hash_builder,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn build(self) -> Cache<K, V, L, S> {
+        let capacity = self.capacity;
+        let shards = self.shards.unwrap_or(Self::DEFAULT_SHARDS);
+        let eviction_config = self.eviction_config.unwrap_or(Self::DEFAULT_EVICTION_CONFIG);
+        let object_pool_capacity = self
+            .object_pool_capacity
+            .unwrap_or(capacity / Self::DEFAULT_OBJECT_POOL_CAPACITY_RATIO_RECIPROCAL);
```

Comment:
Due to `DEFAULT_OBJECT_POOL_CAPACITY_RATIO_RECIPROCAL==10` , the capacity must greater than 9. Otherwise, it will panic due to zero object pool capacity.

Reply:
Fotgot the `AsyncQueue` capacity cannot be 0. Let me open an issue.